lib/active_record_shards/association_scope-4-0.rb

```diff
@@ -0,0 +1,19 @@
+module ActiveRecordShards
+  module AssociationScope
+    def self.included(base)
+      base.send(:alias_method, :bind_without_default_shard, :bind)
+      base.send(:alias_method, :bind, :bind_with_default_shard)
+    end
+
+    def bind_with_default_shard(scope, table_name, column_name, value, tracker)
+      klass = scope.klass
+      if klass.is_sharded? && klass.current_shard_id.nil? && table_name != ActiveRecord::SchemaMigration.table_name
```

Comment:
already have this logic somewhere ... would be nice to just apply that here ... `scope.klass.needs_auto_first_shard?`